src/gluonts/mx/model/deepvar_hierarchical/_network.py

```diff
@@ -216,7 +216,7 @@ def get_samples_for_loss(self, distr: Distribution) -> Tensor:
 
         # Determine which epoch we are currently in.
         self.batch_no += 1
-        epoch_no = self.batch_no // self.num_batches_per_epoch + 1
+        epoch_no = (self.batch_no - 1) // self.num_batches_per_epoch + 1
```

Comment:
Not sure we want to update this one.

Reply:
I think we're currently undercounting the first epoch. 

Assuming batches_per_epoch=10, I think it  goes
Batch no, epoch no
1, 1
2, 1
3, 1
4, 1
5, 1
6, 1
7, 1
8, 1
9, 1
10, 2
11, 2
...